### PR TITLE
Add integration-tests profile

### PR DESCRIPTION
**Pull request by dupliaka**
Runs cypress tests on ERP and VRP if SKIP_INTEGRATION_TESTS id is false

### JIRA

https://issues.redhat.com/browse/PLANNER-2329

### Referenced pull requests
https://github.com/kiegroup/optaweb-vehicle-routing/pull/426
https://github.com/kiegroup/optaweb-employee-rostering/pull/565

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -253,6 +254,20 @@ pipeline {
     }
 }
 
+void saveArtifacts(boolean skipIntegrationTests=false) {
+    if (!skipIntegrationTests) {
+        archiveArtifacts(allowEmptyArchive: true, artifacts: "**/cypress/screenshots/**,**/cypress/videos/**", fingerprint: false)
+    }
```
Comment (radtriste) [inline Jenkinsfile.deploy:260]: you could just have

```suggestion
        archiveArtifacts(allowEmptyArchive: skipIntegrationTests, artifacts: "**/cypress/screenshots/**,**/cypress/videos/**", fingerprint: false)
```

File: Jenkinsfile.deploy [hunk 2]
```diff
@@ -253,6 +254,20 @@ pipeline {
...
+}
+
+List getIntegrationTestProfiles(){
+    integrationTestProfiles = []
+    if (!params.SKIP_INTEGRATION_TESTS) {
+        integrationTestProfiles.add('integration-tests')
+    }
+    return integrationTestProfiles
```
Comment (radtriste) [inline Jenkinsfile.deploy:268]:
```suggestion
    return params.SKIP_INTEGRATION_TESTS ? [] : ['integration-tests']
```

**Comment by triceo** [comment]
I'm not saying no, but I think that we need to talk (@yurloc , @rsynek ?) whether we want this in or not.
This will significantly increase the build time, and raise the chances of flakes.

**Comment by rsynek** [comment]
@triceo I would definitely not make the Cypress tests a part of the release build - a random failure will introduce a delay in the release schedule. If we are enabling these tests, let's do it on nightlies only (so far these run on PR checks and on QE nightlies).

@dupliaka and @yurloc what's your opinion?

**Comment by dupliaka** [comment]
> @triceo I would definitely not make the Cypress tests a part of the release build - a random failure will introduce a delay in the release schedule. If we are enabling these tests, let's do it on nightlies only (so far these run on PR checks and on QE nightlies).
> 
> @dupliaka and @yurloc what's your opinion?
No objections, just got to the stale TODO part and took the initiative to fix that.

**Comment by yurloc** [comment]
> @triceo I would definitely not make the Cypress tests a part of the release build - a random failure will introduce a delay in the release schedule. If we are enabling these tests, let's do it on nightlies only (so far these run on PR checks and on QE nightlies).
> 
> @dupliaka and @yurloc what's your opinion?

I'm not sure about Cypress stability. Personally I don't remember any random failures. Any examples? If not, let's run them always until proven unstable.

If you're aware of any instability, then 1) let's try to make them stable, 2) if that's to expensive, exclude Cypress tests from the release build.

Cypress tests must run on PRs and pushes to long-lived branches. I would keep them in nightly builds and FDB as well.

File: Jenkinsfile.deploy [hunk 3]
```diff
@@ -178,8 +179,23 @@ pipeline {
             post {
                 always {
                     saveReports(params.SKIP_TESTS)
-                    // TODO: Enable together with the profile "integration-tests" once dockerhub pull limit is resolved.
-                    // archiveArtifacts(allowEmptyArchive: true, artifacts: "**/cypress/screenshots/**,**/cypress/videos/**", fingerprint: false)
+                }
+            }
+        }
+
+        stage("Run integration tests Vehicle Routing") {
```
Comment (rsynek) [inline Jenkinsfile.deploy:186]: Running the integration tests in a separate stage means also running the maven build twice, which prolongs the execution time of the pipeline. I would prefer if we could decide whether or not to run the IT in the original `Build ...` stage.

**Review by rsynek — CHANGES_REQUESTED** [review]

File: Jenkinsfile.deploy (hunk 3 above)
Comment (dupliaka) [reply Jenkinsfile.deploy]: Fixed c1b6a5f261ff0b981acce235924b04f0edc575e7

**Pull request closed without merging**